src/store/preferences.ts

```diff
@@ -25,6 +25,10 @@ interface PreferencesState {
   preferencesError: string;
 }
 
+interface CommitArgs extends ServerState {
+  payload?: Partial<Settings>;
```

Comment:
```suggestion
  payload?: RecursivePartial<Settings>;
```
`Partial<{a: {b: 1}}>` does not accept `{a: {}}`, because `Partial` is not recursive (it expects any children that exist to be the same type, and not `Partial` itself).